src/main/java/seedu/address/model/animal/AnimalComparator.java

```diff
@@ -45,4 +30,75 @@ public int compare(Animal o1, Animal o2) {
             }
         }
     };
+
+    private static final Comparator<Animal> ANIMAL_ID_COMPARATOR = new Comparator<Animal>() {
+        @Override
+        public int compare(Animal o1, Animal o2) {
+            return Integer.compare(
+                    Integer.parseInt(o1.getId().value),
+                    Integer.parseInt(o2.getId().value));
+        }
+    };
+
+    private static final Comparator<Animal> ANIMAL_NAME_COMPARATOR = new Comparator<Animal>() {
+        @Override
+        public int compare(Animal o1, Animal o2) {
+            return o1.getName().fullName.compareTo(o2.getName().fullName);
+        }
+    };
+
+    private String category;
```

Comment:
see comments on `SortCommandParser`. i believe category is only used in your `SortCommand` to check if it is an invalid case? in that case `category` attribute can be deleted, then this class won't have to depend on `SortCommandParser`.

Reply:
I think I needed the category for the printing of the success message for the sort command based on the parameter entered.

Reply:
ah okay fair enough!